expression/builtin_time_test.go

```diff
@@ -1284,12 +1287,15 @@ func (s *testEvaluatorSuite) TestStrToDate(c *C) {
 		if !test.Success {
 			c.Assert(err, IsNil)
 			c.Assert(result.IsNull(), IsTrue)
+			s.ctx.GetSessionVars().StmtCtx.ResetForRetry()
 			continue
 		}
 		c.Assert(result.Kind(), Equals, types.KindMysqlTime)
 		value := result.GetMysqlTime()
 		t1, _ := value.Time.GoTime(time.Local)
 		c.Assert(t1, Equals, test.Expect)
+		c.Assert(s.ctx.GetSessionVars().StmtCtx.WarningCount(), Equals, test.Warnings)
```

Comment:
Maybe we should check the content of warnings.

Reply:
Thank you for your review. I'm improving the unit tests now.

Reply:
Hi, Please address this comment when you're free.